server/engine/engine.go

```diff
@@ -222,6 +220,16 @@ func (engine *Engine) GetLatestExecution(step model.Step) model.Execution {
 }
 
 func (engine *Engine) startExecution(step model.Step, execution *model.Execution, waitGroup *sync.WaitGroup) {
+	if step.Name == model.DryRunStepName {
```

Comment:
Shouldn't this block go after the next block to make sure the execution is marked started?

Reply:
The point I was getting at is that we have an established pattern here and the dry run is now doing things differently. 
Having said that, I'm fine with adding this to the list of things to refactor once we get it all running.